css-scroll-snap-2/Overview.bs

```diff
@@ -193,26 +193,117 @@ The 'scroll-start' property {#scroll-start}
 
-  <pre class="propdef shorthand">
-  Name: scroll-start-target
-  Value: [ none | auto ]{1,2}
-  </pre>
-
-  This property is a shorthand property that sets all of the scroll-start-target-* longhands in one declaration.
-  The first value defines the scroll starting point in the block axis,
-  the second sets it in the inline axis.
-  If the second value is omitted, it defaults to ''none''.
-
-  Values are defined as follows:
-
-  <dl dfn-type=value dfn-for="scroll-start-target, scroll-start-target-x, scroll-start-target-y, scroll-start-target-block, scroll-start-target-inline">
-    <dt><dfn>none</dfn>
-    <dd>Element is not a ''scroll-start-target''.
-    <dt><dfn>auto</dfn>
-    <dd>Element is used to calculate the ''scroll-start'' position, 
-    taking into account ''scroll-padding'' or ''scroll-margin'' ,
-    same as a ''scroll-snap'' target.
-  </dl>
+	<h4 id="initial-scroll-target">Initial scroll target</h4>
+
+		The <dfn>initial scroll target</dfn> of a <a>scroll container</a> |scrollcontainer|
+		with respect to a particular axis is an element or pseudo-element whose {{scroll-start-target}}
+		property for that axis is non-''none'' and whose nearest <a>scroll container</a> is
+		|scrollcontainer|. Where multiple such elements or pseudo-elements exist, user-agents
+		should select the one which comes first in <a href="https://www.w3.org/TR/dom/#concept-tree-order">tree order</a>. Where no such element
+		or pseudo-element exists, |scrollcontainer|’s <a>initial scroll target</a> in
+		that axis is null.
+
+		Except for [[#cases-where-scroll-start-target-is-inactive|cases where scroll-start-target is inactive]],
+		user-agents should use the <a>initial scroll target</a> for a <a>scroll container</a> |scrollcontainer|
+		to determine the <a>initial scroll position</a> of |scrollcontainer| along each
+		axis by running the steps to <a>apply initial scroll targets</a>
+
+		The steps to <dfn>apply initial scroll targets</dfn> for a <a>scroll container</a>
+		|scrollcontainer| are as follows:
+
+		1. Let |blocktarget| be the <a>initial scroll target</a> for |scrollcontainer| in the block axis.
+		1. Let |inlinetarget| be the <a>initial scroll target</a> for |scrollcontainer| in the inline axis.
+		1. If |blocktarget| is not null:
+			1. Set the <a>block axis</a> component of |scrollcontainer|’s <a>initial scroll position</a>
+				to the result of running the steps to <a>get initial scroll offset</a> for
+				|scrollcontainer| with the axis parameter as “block” and the target parameter
+				as |blocktarget|.
+		1. If |inlinetarget| is not null:
+			1. Set the <a>inline axis</a> component of |scrollcontainer|’s <a>initial scroll position</a>
+				to the result of running the steps to <a>get initial scroll offset</a> for
+				|scrollcontainer| with the axis parameter as “inline” and the target parameter
+				as |inlinetarget|.
+
+		The steps to <dfn>get initial scroll offset</dfn> for a <a>scroll container</a> |scrollcontianer|,
```

Comment:
I think this should be calling into a common algorithm used by the spec rules for [scrolling an element into view](https://drafts.csswg.org/cssom-view/#scroll-a-target-into-view) as we would expect this to behave the same position as you would get by calling scrollIntoView.

Reply:
I think we intend for scroll-start-target to only affect its nearest scroll ancestor. Invoking "[scrolling an element into view](https://drafts.csswg.org/cssom-view/#scroll-a-target-into-view)" as written would result in the scrolling of all ancestor scrolling boxes. I've modified it to define an optional _nearest_only_ parameter for this algorithm which will default to false.